pallets/msa/src/lib.rs

```diff
@@ -645,6 +672,37 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	/// Disables given MSA by expiring it at the current block.
+	///
+	/// # Arguments
+	/// * `msa_id` - The user's MSA to be retired.
+	///
+	/// # Returns
+	/// * [`DispatchResult`]
+	///
+	/// # Errors
+	/// * [`Error::<T>::_`] - MSA is already retired
+	pub fn retire_msa(
+		msa_id: MessageSourceId
+	) -> DispatchResult {
+
+		ProviderInfoOf::<T>::iter_key_prefix(msa_id.into())
+			.for_each(|delegator: Delegator| -> Result<(), Error<T>> {
+				Self::revoke_provider(msa_id.into(), delegator.into()).map_err(|_| Error::<T>::MsaRevoked)
+			});
```

Comment:
There needs to be a different error handling here, because right now the operation fails at the first error, instead of continuing the loop. I need to handle the closure of 'for each' to skip forward.
- Do we need to throw an error here in case there are any errors with revoking delegations at all? or can those errors be swallowed?

Reply:
Why you are not using a regular loop instead of for_each?